src/Ledger/Utxo/Properties.lagda

```diff
@@ -544,3 +548,194 @@ pov {tx}{utxo}{_}{fees}{deposits}{donations} h'
 \end{code}
 
 \end{property}
+
+\begin{property}[\textbf{General Minimum Spending Condition}]~\\
+
+\begin{code}[hide]
+isRefundCert : DCert → Bool
+isRefundCert (dereg c) = true
+isRefundCert (deregdrep c) = true
+isRefundCert _ = false
+
+noRefundCert : List DCert → Set _
+noRefundCert l = All (λ cert → isRefundCert cert ≡ false) l
+
+opaque
+  unfolding List-Model
+  unfolding finiteness
+  fin∘list[] : {A : Set} → proj₁ (finiteness{A = A} ∅) ≡ []
+  fin∘list[] = refl
+  fin∘list∷[] : {A : Set} {a : A} → proj₁ (finiteness ❴ a ❵) ≡ [ a ]
+  fin∘list∷[] = refl
+
+coin∅ : getCoin{A = DepositPurpose ⇀ Coin} ∅ ≡ 0
+coin∅ = begin
+  foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ (proj₁ (finiteness ∅)))
+    ≡⟨ cong (λ u → (foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ u))) fin∘list[] ⟩
+  foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0 (deduplicate _≟_ [])
+    ≡⟨ cong (λ u → (foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0  u))
+            {x = deduplicate _≟_ []} {y = []} refl ⟩
+  foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0 []
+    ≡⟨ refl ⟩
+  0 ∎
+  where open Prelude.≡-Reasoning
+
+getCoin-singleton : ((dp , c) : DepositPurpose × Coin) → indexedSumᵛ' id ❴ (dp , c) ❵ ≡ c
+getCoin-singleton (dp , c) = begin
+  foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ (proj₁ (finiteness (proj₁ ❴ (dp , c) ❵))))
+    ≡⟨ cong (λ u → foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ u)) fin∘list∷[] ⟩
+  foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ ((dp , c) ∷ []))
+    ≡⟨ cong (λ u → foldr (λ x → (proj₂ x) +_) 0 u)
+            {x = deduplicate _≟_ ((dp , c) ∷ [])} {y = (dp , c) ∷ []} refl ⟩
+  foldr (λ x → (proj₂ x) +_) 0 ((dp , c) ∷ [])
+    ≡⟨ +-identityʳ c ⟩
+  c ∎
+  where open Prelude.≡-Reasoning
+
+-- ⊆→∪ : (d d' : ℙ (DepositPurpose × Coin)) → d ⊆ d' → d' ≡ᵉ d' ∪ d
+-- ⊆→∪ d d' d⊆d' = ∪-⊆ˡ , λ {a} x → case from ∈-∪ x of λ where
+--   (inj₁ v) → v
+--   (inj₂ v) → d⊆d' v
+
+module _
+  -- ASSUMPTION --
+  {- 1 -} {gc-hom : (d₁ d₂ : DepositPurpose ⇀ Coin) → getCoin (d₁ ∪⁺ d₂) ≡ getCoin d₁ + getCoin d₂}
+  where
+  ∪⁺∅ᵐ≡id : ∀ {d} → getCoin d ≡ getCoin (d ∪⁺ ∅ᵐ)
```

Comment:
The reason I changed this proof to inline is (a) it's a minor lemma and (b) the proof gets uglier with `≡-Reasoning` because it requires specifying implicit types for `getCoin ∅`:

```agda
  ∪⁺∅ᵐ≡id : ∀ {d} → getCoin d ≡ getCoin (d ∪⁺ ∅ᵐ)
  ∪⁺∅ᵐ≡id {d} = begin
    getCoin d
      ≡⟨ sym (+-identityʳ (getCoin d)) ⟩
    getCoin d + 0
      ≡⟨ cong (getCoin d +_) (sym coin∅) ⟩
    getCoin d + getCoin{A = DepositPurpose ⇀ Coin} ∅   --  <- here
      ≡⟨ sym (gc-hom d ∅) ⟩
    getCoin (d ∪⁺ ∅ᵐ)
      ∎
    where open Prelude.≡-Reasoning
```